### PR TITLE
UIREQ-700: Select item modal should display all items in the Instance…

### PR DESCRIPTION
**Pull request by pavelspichonak**
## Purpose
Select item modal should display all items in the Instance when user uncheck box

## Approach
Now we have **ItemsDialog** component which just render passed items.

This component is used in **move request dialog** and in **create request form**. Both these places get items from backend but **create request form** does not get all necessary items information(it is a bug) but **move request dialog** get all necessary items information.

So we decided:
1) remove all logic from **move request dialog** and **create request form** related to getting items
2) move this logic to **ItemsDialog** component(using correct logic from **move request dialog**)
3) add items sorting based on discussion with PO that items with status available must go first

## Refs
https://issues.folio.org/browse/UIREQ-700

### CODE REVIEW

**Review by Stanislau-Snapok — APPROVED** [review]

**Comment by pavelspichonak** [comment]
@zburke @mkuklis could you please review?

File: src/ItemsDialog.js [hunk 1]
```diff
@@ -58,16 +65,102 @@ export const formatter = {
...
+  const contentData = useMemo(() => {
+    let resultItems = items;
+
+    if (skippedItemId) {
+      resultItems = resultItems
+        .filter(item => skippedItemId !== item.id);
+    }
+
+    // items with status available must go first
+    return resultItems.sort((a, b) => {
```
Comment (mkuklis) [inline src/ItemsDialog.js:152]: @pavelspichonak I think you can shorter this with:

````js
return resultItems.sort(a => a?.status?.name === itemStatuses.AVAILABLE ? -1 : 1);
````

**Review by mkuklis — APPROVED** [review]
@pavelspichonak nice and clean refactoring! 

I just left a tiny little comment.

File: src/ItemsDialog.js (hunk 1 above)
Comment (pavelspichonak) [reply src/ItemsDialog.js]: Done. Thanks

**Pull request merged**